olive/systems/azureml/aml_system.py

```diff
@@ -115,14 +117,104 @@ def __init__(
         self.env_vars = self._get_hf_token_env(self.azureml_client_config.keyvault_name) if self.hf_token else None
         self.temp_dirs = []
 
+    def submit_workflow(self, run_config: RunConfig):
+        workflow_id = run_config.workflow_id
+        logger.info("Submitting workflow %s to the AzureML system.", workflow_id)
+        ml_client = self.azureml_client_config.create_client()
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            workflow_pipeline = self._create_pipeline_for_workflow(run_config, tempdir)
+            self._run_workflow_job(
+                ml_client,
+                workflow_pipeline,
+                "olive-workflow",
+                tags={"Workflow": workflow_id},
+            )
+
+    def _create_pipeline_for_workflow(self, run_config: RunConfig, tmp_dir):
+        workflow_id = run_config.workflow_id
+        logger.info("Creating pipeline for workflow %s", workflow_id)
+        script_name = "aml_workflow_runner.py"
+
+        tmp_dir = Path(tmp_dir)
+
+        run_config.workflow_host = None
+        run_config.engine.log_to_file = True
+        olive_config = run_config.to_json()
+
+        olive_config_path = self._create_olive_config_file(olive_config, tmp_dir)
+
+        config_root = tmp_dir / "config"
+        cur_dir = Path(__file__).resolve().parent
+        code_files = [cur_dir / script_name, olive_config_path]
+        self.copy_files(code_files, config_root)
+
+        inputs = {WORKFLOW_CONFIG: Input(type=AssetTypes.URI_FILE)}
+        args = {WORKFLOW_CONFIG: Input(type=AssetTypes.URI_FILE, path=olive_config_path)}
+
+        config_list = []
+        config_list.append(("model", run_config.input_model.to_json(check_object=True)))
+
+        if run_config.data_configs:
+            config_list.extend(
+                [(data_config.name, data_config.to_json(check_object=True)) for data_config in run_config.data_configs]
+            )
+
+        resources_check_list = [run_config.passes, run_config.evaluators, run_config.systems]
+        for resources in resources_check_list:
+            if resources:
+                config_list = self._get_dict_resources_config(resources, config_list)
+
+        for name, config_json in config_list:
+            name_inputs, name_args = self.create_inputs_and_args(
```

Comment:
`create_inputs_and_args` has been updated to take all configs at once to avoid duplicate mounts. 

It should be possible to just do 
`inputs, args = self.create_inputs_and_args({"run_config" : run_config.to_json()}, "tmp_dir", ignore_keys=["engine"m "model_attributes"]` 

Similarly, in the aml workflow runner it can just parse the run_config itself at once.

Reply:
although I am not sure about ignoring the engine entirely since it must have been valided to replace evaluators with the full configuration.

Reply:
Will update this.